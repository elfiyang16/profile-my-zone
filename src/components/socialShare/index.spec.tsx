import { shallow, mount } from "enzyme"
import React from "react"
import { axe, toHaveNoViolations } from "jest-axe"
import SocialSharer from "./"

expect.extend(toHaveNoViolations)

describe("SocialSharer", () => {
  it("should match the latest snapshot", () => {
    expect(
      shallow(<SocialSharer blogTitle={"mockPost"} blogSlug={"/mockPost"} />)
    ).toMatchSnapshot()
  })

  it("should check accessibility", async () => {
    const wrapper = mount(
      <SocialSharer blogTitle={"mockPost"} blogSlug={"/mockPost"} />
    )
    const results = await axe(wrapper.getDOMNode())

    expect(results).toHaveNoViolations()
  })
})
