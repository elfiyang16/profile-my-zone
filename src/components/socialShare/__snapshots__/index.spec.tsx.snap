// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialSharer should match the latest snapshot 1`] = `
<div
  aria-label="Social sharing button"
  className="social-share"
  role="region"
>
  <p
    className="social-share_wrapper"
  >
    Share on:
     
    <a
      className="social-share__link"
      href="https://twitter.com/intent/tweet/?text=mockPost&url=false&via=elfi"
      rel="nofollow"
      target="_blank"
      title="Share on Twitter"
    >
      <img
        alt="Twitter"
        className="social-share__icon"
        src={Object {}}
      />
    </a>
  </p>
</div>
`;
